contracts/liquidity_hub/pool-manager/src/helpers.rs

```diff
@@ -174,26 +174,61 @@ pub fn compute_swap(
             let burn_fee_amount: Uint256 = pool_fees.burn_fee.compute(return_amount);
 
             // swap and protocol fee will be absorbed by the pool. Burn fee amount will be burned on a subsequent msg.
-            let return_amount: Uint256 =
-                return_amount - swap_fee_amount - protocol_fee_amount - burn_fee_amount;
+            #[cfg(not(feature = "osmosis"))]
+            {
+                let return_amount: Uint256 =
+                    return_amount - swap_fee_amount - protocol_fee_amount - burn_fee_amount;
+
+                Ok(SwapComputation {
+                    return_amount: return_amount
+                        .try_into()
+                        .map_err(|_| ContractError::SwapOverflowError {})?,
+                    spread_amount: spread_amount
+                        .try_into()
+                        .map_err(|_| ContractError::SwapOverflowError {})?,
+                    swap_fee_amount: swap_fee_amount
+                        .try_into()
+                        .map_err(|_| ContractError::SwapOverflowError {})?,
+                    protocol_fee_amount: protocol_fee_amount
+                        .try_into()
+                        .map_err(|_| ContractError::SwapOverflowError {})?,
+                    burn_fee_amount: burn_fee_amount
+                        .try_into()
+                        .map_err(|_| ContractError::SwapOverflowError {})?,
+                })
+            }
 
-            Ok(SwapComputation {
-                return_amount: return_amount
-                    .try_into()
-                    .map_err(|_| ContractError::SwapOverflowError {})?,
-                spread_amount: spread_amount
-                    .try_into()
-                    .map_err(|_| ContractError::SwapOverflowError {})?,
-                swap_fee_amount: swap_fee_amount
-                    .try_into()
-                    .map_err(|_| ContractError::SwapOverflowError {})?,
-                protocol_fee_amount: protocol_fee_amount
-                    .try_into()
-                    .map_err(|_| ContractError::SwapOverflowError {})?,
-                burn_fee_amount: burn_fee_amount
-                    .try_into()
-                    .map_err(|_| ContractError::SwapOverflowError {})?,
-            })
+            #[cfg(feature = "osmosis")]
+            {
+                let osmosis_fee_amount: Uint256 = pool_fees.osmosis_fee.compute(return_amount);
+
+                let return_amount: Uint256 = return_amount
+                    - swap_fee_amount
```

Comment:
better use safe math here

Reply:
yep, good point

Reply:
I think it was done that way since the return_amount is bigger than the subsequent values, as they are a % of the return_amount. Safe enough, but will change it thx for flagging it.